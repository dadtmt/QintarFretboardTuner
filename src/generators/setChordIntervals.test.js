import { generateFretboard, setChordIntervals } from './index'

describe('setChordIntervals C3maj', () => {
  it('sets the tonic', () => {
    const fretboard = generateFretboard('F1')
    const indexC3 = 44
    const maj = { name: 'major', intervals: ['1P', '3M', '5P'] }
    expect(setChordIntervals(fretboard, indexC3, maj)).toMatchSnapshot()
  })
})
