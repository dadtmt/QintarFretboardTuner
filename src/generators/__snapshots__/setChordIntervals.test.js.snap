// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`setChordIntervals C3maj sets the tonic 1`] = `
Array [
  Object {
    "fret": "E4",
    "fretIndex": 0,
    "selectedChordInterval": "",
  },
  Object {
    "fret": "F4",
    "fretIndex": 1,
    "selectedChordInterval": "",
  },
  Object {
    "fret": "Gb4",
    "fretIndex": 2,
    "selectedChordInterval": "",
  },
  Object {
    "fret": "G4",
    "fretIndex": 3,
    "selectedChordInterval": "",
  },
  Object {
    "fret": "Ab4",
    "fretIndex": 4,
    "selectedChordInterval": "",
  },
  Object {
    "fret": "A4",
    "fretIndex": 5,
    "selectedChordInterval": "",
  },
  Object {
    "fret": "A#4",
    "fretIndex": 6,
    "selectedChordInterval": "",
  },
  Object {
    "fret": "B4",
    "fretIndex": 7,
    "selectedChordInterval": "",
  },
  Object {
    "fret": "C5",
    "fretIndex": 8,
    "selectedChordInterval": "",
  },
  Object {
    "fret": "Db5",
    "fretIndex": 9,
    "selectedChordInterval": "",
  },
  Object {
    "fret": "D5",
    "fretIndex": 10,
    "selectedChordInterval": "",
  },
  Object {
    "fret": "Eb5",
    "fretIndex": 11,
    "selectedChordInterval": "",
  },
  Object {
    "fret": "E5",
    "fretIndex": 12,
    "selectedChordInterval": "",
  },
  Object {
    "fret": "A3",
    "fretIndex": 13,
    "selectedChordInterval": "",
  },
  Object {
    "fret": "A#3",
    "fretIndex": 14,
    "selectedChordInterval": "",
  },
  Object {
    "fret": "B3",
    "fretIndex": 15,
    "selectedChordInterval": "",
  },
  Object {
    "fret": "C4",
    "fretIndex": 16,
    "selectedChordInterval": "",
  },
  Object {
    "fret": "Db4",
    "fretIndex": 17,
    "selectedChordInterval": "",
  },
  Object {
    "fret": "D4",
    "fretIndex": 18,
    "selectedChordInterval": "",
  },
  Object {
    "fret": "D#4",
    "fretIndex": 19,
    "selectedChordInterval": "",
  },
  Object {
    "fret": "E4",
    "fretIndex": 20,
    "selectedChordInterval": "",
  },
  Object {
    "fret": "F4",
    "fretIndex": 21,
    "selectedChordInterval": "",
  },
  Object {
    "fret": "Gb4",
    "fretIndex": 22,
    "selectedChordInterval": "",
  },
  Object {
    "fret": "G4",
    "fretIndex": 23,
    "selectedChordInterval": "",
  },
  Object {
    "fret": "Ab4",
    "fretIndex": 24,
    "selectedChordInterval": "",
  },
  Object {
    "fret": "A4",
    "fretIndex": 25,
    "selectedChordInterval": "",
  },
  Object {
    "fret": "D3",
    "fretIndex": 26,
    "selectedChordInterval": "",
  },
  Object {
    "fret": "D#3",
    "fretIndex": 27,
    "selectedChordInterval": "",
  },
  Object {
    "fret": "E3",
    "fretIndex": 28,
    "selectedChordInterval": "",
  },
  Object {
    "fret": "F3",
    "fretIndex": 29,
    "selectedChordInterval": "",
  },
  Object {
    "fret": "Gb3",
    "fretIndex": 30,
    "selectedChordInterval": "",
  },
  Object {
    "fret": "G3",
    "fretIndex": 31,
    "selectedChordInterval": "",
  },
  Object {
    "fret": "G#3",
    "fretIndex": 32,
    "selectedChordInterval": "",
  },
  Object {
    "fret": "A3",
    "fretIndex": 33,
    "selectedChordInterval": "",
  },
  Object {
    "fret": "A#3",
    "fretIndex": 34,
    "selectedChordInterval": "",
  },
  Object {
    "fret": "B3",
    "fretIndex": 35,
    "selectedChordInterval": "",
  },
  Object {
    "fret": "C4",
    "fretIndex": 36,
    "selectedChordInterval": "",
  },
  Object {
    "fret": "Db4",
    "fretIndex": 37,
    "selectedChordInterval": "",
  },
  Object {
    "fret": "D4",
    "fretIndex": 38,
    "selectedChordInterval": "",
  },
  Object {
    "fret": "G2",
    "fretIndex": 39,
    "selectedChordInterval": "",
  },
  Object {
    "fret": "G#2",
    "fretIndex": 40,
    "selectedChordInterval": "",
  },
  Object {
    "fret": "A2",
    "fretIndex": 41,
    "selectedChordInterval": "",
  },
  Object {
    "fret": "A#2",
    "fretIndex": 42,
    "selectedChordInterval": "",
  },
  Object {
    "fret": "B2",
    "fretIndex": 43,
    "selectedChordInterval": "",
  },
  Object {
    "fret": "C3",
    "fretIndex": 44,
    "selectedChordInterval": "1P",
  },
  Object {
    "fret": "C#3",
    "fretIndex": 45,
    "selectedChordInterval": "",
  },
  Object {
    "fret": "D3",
    "fretIndex": 46,
    "selectedChordInterval": "",
  },
  Object {
    "fret": "D#3",
    "fretIndex": 47,
    "selectedChordInterval": "",
  },
  Object {
    "fret": "E3",
    "fretIndex": 48,
    "selectedChordInterval": "",
  },
  Object {
    "fret": "F3",
    "fretIndex": 49,
    "selectedChordInterval": "",
  },
  Object {
    "fret": "Gb3",
    "fretIndex": 50,
    "selectedChordInterval": "",
  },
  Object {
    "fret": "G3",
    "fretIndex": 51,
    "selectedChordInterval": "",
  },
  Object {
    "fret": "C2",
    "fretIndex": 52,
    "selectedChordInterval": "",
  },
  Object {
    "fret": "C#2",
    "fretIndex": 53,
    "selectedChordInterval": "",
  },
  Object {
    "fret": "D2",
    "fretIndex": 54,
    "selectedChordInterval": "",
  },
  Object {
    "fret": "D#2",
    "fretIndex": 55,
    "selectedChordInterval": "",
  },
  Object {
    "fret": "E2",
    "fretIndex": 56,
    "selectedChordInterval": "",
  },
  Object {
    "fret": "F2",
    "fretIndex": 57,
    "selectedChordInterval": "",
  },
  Object {
    "fret": "F#2",
    "fretIndex": 58,
    "selectedChordInterval": "",
  },
  Object {
    "fret": "G2",
    "fretIndex": 59,
    "selectedChordInterval": "",
  },
  Object {
    "fret": "G#2",
    "fretIndex": 60,
    "selectedChordInterval": "",
  },
  Object {
    "fret": "A2",
    "fretIndex": 61,
    "selectedChordInterval": "",
  },
  Object {
    "fret": "A#2",
    "fretIndex": 62,
    "selectedChordInterval": "",
  },
  Object {
    "fret": "B2",
    "fretIndex": 63,
    "selectedChordInterval": "",
  },
  Object {
    "fret": "C3",
    "fretIndex": 64,
    "selectedChordInterval": "",
  },
  Object {
    "fret": "F1",
    "fretIndex": 65,
    "selectedChordInterval": "",
  },
  Object {
    "fret": "F#1",
    "fretIndex": 66,
    "selectedChordInterval": "",
  },
  Object {
    "fret": "G1",
    "fretIndex": 67,
    "selectedChordInterval": "",
  },
  Object {
    "fret": "G#1",
    "fretIndex": 68,
    "selectedChordInterval": "",
  },
  Object {
    "fret": "A1",
    "fretIndex": 69,
    "selectedChordInterval": "",
  },
  Object {
    "fret": "A#1",
    "fretIndex": 70,
    "selectedChordInterval": "",
  },
  Object {
    "fret": "B1",
    "fretIndex": 71,
    "selectedChordInterval": "",
  },
  Object {
    "fret": "C2",
    "fretIndex": 72,
    "selectedChordInterval": "",
  },
  Object {
    "fret": "C#2",
    "fretIndex": 73,
    "selectedChordInterval": "",
  },
  Object {
    "fret": "D2",
    "fretIndex": 74,
    "selectedChordInterval": "",
  },
  Object {
    "fret": "D#2",
    "fretIndex": 75,
    "selectedChordInterval": "",
  },
  Object {
    "fret": "E2",
    "fretIndex": 76,
    "selectedChordInterval": "",
  },
  Object {
    "fret": "F2",
    "fretIndex": 77,
    "selectedChordInterval": "",
  },
]
`;
